pkg/common/helper_test.go

```diff
@@ -0,0 +1,167 @@
+package common
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/suite"
+)
+
+type IsFileTestSuite struct {
+	suite.Suite
+	tempDir string
+}
+
+type IsDirTestSuite struct {
+	suite.Suite
+	tempDir string
+}
+
+type FileExistTestSuite struct {
+	suite.Suite
+	tempDir string
+}
+
+type StringSliceToIntSliceTestSuite struct {
+	suite.Suite
+}
+
+type RetryUntilSuccessfulTestSuite struct {
+	suite.Suite
+}
+
+func (suite *IsFileTestSuite) SetupTest()  {
+	var err error
+
+	// create a temp directory
+	suite.tempDir, err = ioutil.TempDir("", "isfile-test")
+	suite.Require().NoError(err)
+}
+
+func (suite *IsFileTestSuite) TestIsFileFuncPositive() {
+	//create temp file
+	tempFile, err := ioutil.TempFile(suite.tempDir, "temp_file")
+	suite.Require().NoError(err)
+	suite.Assert().True(IsFile(tempFile.Name()))
+	defer os.Remove(tempFile.Name())
+}
+
+func (suite *IsFileTestSuite) TestIsFileFuncFileIsNotExist() {
+	//create temp file
+	tempFile, err := ioutil.TempFile(suite.tempDir, "temp_file")
+	suite.Require().NoError(err)
+	os.Remove(tempFile.Name())
+	suite.Assert().False(IsFile(tempFile.Name()))
+}
+
+func (suite *IsFileTestSuite) TestIsFileFuncFileIsADirectory() {
+	suite.Assert().False(IsFile(suite.tempDir))
+}
+
+func (suite *IsFileTestSuite) TearDown() {
+	defer os.RemoveAll(suite.tempDir)
+}
+
+func (suite *IsDirTestSuite) TestIsDirFuncPositive() {
+	var err error
+
+	// create a temp directory
+	suite.tempDir, err = ioutil.TempDir("", "isdir-test")
```

Comment:
1. Cleanup?
2. Here you set `suite.tempDir` but don't use it outside the function. Don't set variables outside of the scope you use them (this should be a local variable)